deepchecks/nlp/checks/model_evaluation/train_test_performance.py

```diff
@@ -0,0 +1,212 @@
+from deepchecks.nlp import Context, TrainTestCheck
+from deepchecks.nlp.metric_utils.scorers import infer_on_text_data
+from deepchecks.nlp.task_type import TaskType
+from deepchecks.nlp.text_data import TextData
+from deepchecks.utils.abstracts.train_test_performace import TrainTestPerformanceAbstract
+from deepchecks.utils.plot import DEFAULT_DATASET_NAMES, colors
+
+__all__ = ['TrainTestPerformance']
+
+
+class TrainTestPerformance(TrainTestPerformanceAbstract, TrainTestCheck):
+    """Summarize given model performance on the train and test datasets based on selected scorers.
+
+    Parameters
+    ----------
+    scorers: Union[Mapping[str, Union[str, Callable]], List[str]], default: None
+        Scorers to override the default scorers, find more about the supported formats at
+        https://docs.deepchecks.com/stable/user-guide/general/metrics_guide.html
+    n_samples : int , default: 1_000_000
+        number of samples to use for this check.
+    random_state : int, default: 42
+        random seed for all check internals.
+
+    Notes
+    -----
+    Scorers are a convention of sklearn to evaluate a model.
+    `See scorers documentation <https://scikit-learn.org/stable/modules/model_evaluation.html#scoring>`__
+    A scorer is a function which accepts (model, X, y_true) and returns a float result which is the score.
+    For every scorer higher scores are better than lower scores.
+
+    You can create a scorer out of existing sklearn metrics:
+
+    .. code-block:: python
+
+        from sklearn.metrics import roc_auc_score, make_scorer
+
+        training_labels = [1, 2, 3]
+        auc_scorer = make_scorer(roc_auc_score, labels=training_labels, multi_class='ovr')
+        # Note that the labels parameter is required for multi-class classification in metrics like roc_auc_score or
+        # log_loss that use the predict_proba function of the model, in case that not all labels are present in the test
+        # set.
+
+    Or you can implement your own:
+
+    .. code-block:: python
+
+        from sklearn.metrics import make_scorer
+
+
+        def my_mse(y_true, y_pred):
+            return (y_true - y_pred) ** 2
+
+
+        # Mark greater_is_better=False, since scorers always suppose to return
+        # value to maximize.
+        my_mse_scorer = make_scorer(my_mse, greater_is_better=False)
+    """
+
+    def __init__(
+        self,
+        scorers: t.Union[
+            t.Mapping[str, t.Union[str, t.Callable]],
+            t.List[str],
+            None
+        ] = None,
+        n_samples: int = 1_000_000,
+        random_state: int = 42,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.scorers = scorers
+        self.n_samples = n_samples
+        self.random_state = random_state
+
+    @classmethod
+    def _default_per_class_scorers(cls) -> t.Dict[str, str]:
+        # TODO:
+        return {}
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Run check."""
```

Comment:
Try to move more of this section into the Abstract - similar to what happening in PredictionDriftAbstract - a function the return CheckResult
Optimal - send scorers and the dataset and do everything there
Minimal - create the display in the abstract (send the scorers results)

Reply:
Yea ok